### ISSUE TITLE
Untrusted SSL cert == "unable to ping"

### ISSUE THREAD

**Issue by wsanchez** [burningmantech/ranger-ims-mac#36]
If the server is using an untrusted X.509 certificate (eg. a self-signed cert), then the application brings up an error dialog when attempting to connect:

![screen shot 2014-07-29 at 10 41 22 am](https://cloud.githubusercontent.com/assets/50002/3739093/bd3ab2c0-1747-11e4-8645-2dd28a7421fc.png)

**Comment by wsanchez** [comment]
This no longer happens, though that's because it's now not validating the cert, so that's kinda dumb.
Filed #49 for that.

**Comment by wsanchez** [comment]
OK, fixed #49, and this is back. Re-opening.
Work-around is to visit the server with (eg.) Safari and accept the cert there; then the IMS app will work.

**Comment by wsanchez** [comment]
Whoops, never mind.  Fixed #62 and now this is back, but #49 is the validation fix, so this can stay closed.

**Comment by wsanchez** [comment]
ad72af44cb1d869dc62155efcce01a0c2775f4cd adds code that will bypass the TLS auth, but that's back to being lame.

What I'd like is the "trust this cert?" dialog one sees in Safari, I think.

Alternatively, we could add a preference to import a trusted CA cert to the app and use certs by that CA (or import a specific cert to trust, but that's less flexible and not any easier on the user).

**Comment by wsanchez** [comment]
Here we go: https://developer.apple.com/library/ios/technotes/tn2232/_index.html#//apple_ref/doc/uid/DTS40012884-CH1-SECTRUSTEXCEPTIONS

**Comment by flwyd** [comment]
Rather than using a self-signed cert, would a free, trusted, cert from https://letsencrypt.org/ solve the problem?

**Comment by flwyd** [comment]
(The actual-cert solution might be tricky if the client is connecting to a server's LAN address rather than a public URL.)

**Comment by wsanchez** [comment]
Yeah, Let's Encrypt would make getting a "real" cert easier, as I think their CA should be in Apple root CA list now, but I'd still like to figure out how to get allow the client to view and accept a random cert, because I think that may be necessary on playa.

Self-signed certs are basically like SSH host keys.  Trust once (ideally verify manually first), then complain when it changes.  I think that model works OK in our usage.
